presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -445,6 +445,10 @@ public static Slice multiply(Slice left, Slice right)
 
     public static void multiply(Slice left, Slice right, Slice result)
     {
+        boolean fourIntsResultExpected = result.length() == NUMBER_OF_LONGS * Long.BYTES;
```

Comment:
We may want to duplicate it anyway. As a I commented elsewhere, we may be able to exploit certain invariants to remove some of the computations and speed up the 16-byte case.